### PR TITLE
Planning: prevented stucking at stage_adjust in Park_and_Go scenario.

### PR DESCRIPTION
**Pull request by sjiang2018**

### CODE REVIEW

File: modules/planning/scenarios/park_and_go/stage_adjust.cc [hunk 1]
```diff
@@ -48,10 +48,22 @@ Stage::StageStatus ParkAndGoStageAdjust::Process(
     AERROR << "ParkAndGoStageAdjust planning error";
     return StageStatus::ERROR;
   }
-  const bool ready_to_cruise =
+  bool is_ready_to_cruise =
```
Comment (xiaoxq) [inline modules/planning/scenarios/park_and_go/stage_adjust.cc]: Keep `const`?

File: modules/planning/scenarios/park_and_go/stage_adjust.cc [hunk 2]
```diff
@@ -48,10 +48,22 @@ Stage::StageStatus ParkAndGoStageAdjust::Process(
...
-  if (!ready_to_cruise) {
+  bool is_end_of_trajectory = false;
+  const auto& history_frame = FrameHistory::Instance()->Latest();
+  if (history_frame) {
+    const auto& trajectory_points =
+        history_frame->current_frame_planned_trajectory().trajectory_point();
+    const auto trajectory_point_size =
+        history_frame->current_frame_planned_trajectory()
+            .trajectory_point_size();
+    auto last_point = trajectory_points[trajectory_point_size - 1];
```
Comment (xiaoxq) [inline modules/planning/scenarios/park_and_go/stage_adjust.cc]: 1.  `const auto&`
2. Simply use `trajectory_points.rbegin()->relative_time()`

File: modules/planning/scenarios/park_and_go/stage_adjust.cc (hunk 1 above)
Comment (sjiang2018) [reply modules/planning/scenarios/park_and_go/stage_adjust.cc]: I thought you said "Do not use const bool" before.

Comment (sjiang2018) [reply modules/planning/scenarios/park_and_go/stage_adjust.cc]: Fixed. Thank you!

File: modules/planning/scenarios/park_and_go/stage_adjust.cc (hunk 2 above)
Comment (sjiang2018) [reply modules/planning/scenarios/park_and_go/stage_adjust.cc]: Fixed. Thank you!

**Pull request merged**